Dockerfile

```diff
@@ -6,7 +6,16 @@
 # docker build -t nav2:latest \
 #   --build-arg UNDERLAY_MIXINS \
 #   --build-arg OVERLAY_MIXINS ./
-ARG FROM_IMAGE=osrf/ros2:nightly
+ARG FROM_IMAGE=osrf/ros2:old-nightly
+FROM $FROM_IMAGE as package_cache
+
+WORKDIR /tmp
+COPY ./ ./src
+RUN mkdir ./cache && \
+    cd ./src && \
+    find ./ -name "package.xml" | \
+    xargs cp --parents -t ../cache
```

Comment:
It looks like you are copying the package.xml files without regard to whether the package is COLCON_IGNOREd. 

The nav2_experimental package is currently COLCON_IGNOREd because certain dependencies are not properly integrated into rosdep yet. Tensor flow and keras are some of its dependencies.

Looks like you are pulling it's package.xml into the cache and then using it to do the rosdep command below.

Reply:
I think you could use the `colcon list -p` command to get the paths of valid packages. However, it's output is a bit different than find, you might have to do some extra transformations

Reply:
Thanks @crdelsey , didn't think to check if COLCON_IGNORE was still used anywhere.
I've simply just added that file name to the copy to cache step in the multi stage.
Works like a charm now, i.e. only package.xml changes break the underlay dependency install.